### PR TITLE
[Src] Template for tensor source

### PR DESCRIPTION
**Pull request by kparichay**
Added empty template for tensor source plugin for linux IIO
Properties for IIO devices have been added, but not yet finalized

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/tensor_source/tensor_src_iio.c [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+#define MAX_BUFFER_CAPACITY 100
+#define DEFAULT_BUFFER_CAPACITY 1
+
+/**
+ * @brief Minimum and maximum operating frequency for the device
+ * Default frequency chooses the first available frequency supported by device
+ */
+#define MIN_FREQUENCY 1
+#define MAX_FREQUENCY 999999999
+#define DEFAULT_FREQUENCY 0
```
Comment (wooksong) [inline gst/nnstreamer/tensor_source/tensor_src_iio.c]: Can DEFAULT_FREQUENCY be less then MIN_FREQUENCY? Even thought this default value is dummy, it looks illogical to me.

File: gst/nnstreamer/tensor_source/tensor_src_iio.c [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+
+/**
+ * @brief Template for src pad.
+ */
+static GstStaticPadTemplate src_factory = GST_STATIC_PAD_TEMPLATE ("src",
+    GST_PAD_SRC,
+    GST_PAD_ALWAYS,
+    GST_STATIC_CAPS (GST_TENSOR_CAP_DEFAULT "; " GST_TENSORS_CAP_DEFAULT));
```
Comment (wooksong) [inline gst/nnstreamer/tensor_source/tensor_src_iio.c]: Can this src generate other/tensors? Do you have any use cases? 
IMO, the following diagram is more general to me....but anyway we do not decide this now.
```bash
tensor_src ---
                    | ------------ tensor_mux (or something else)
tensor_src ---
```

File: gst/nnstreamer/tensor_source/tensor_src_iio.c (hunk 1 above)
Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: The default frequency is set to 0 indicating that user has not specified any specific frequency and program will use the first value from the list of the available frequency supported.

File: gst/nnstreamer/tensor_source/tensor_src_iio.c (hunk 2 above)
Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: I agree with you, other/tensors would not be needed for direct output from the tensor_src_iio and I donot have a proper use case for it.
However, it seems that multiple channels can have different sampling frequencies with buffered data (I am not sure about it, I have yet to find a proper documentation for this but I saw an example for this online). So, other/tensors might be needed but not sure yet.

**Pull request merged**